sharktank/tests/evaluate/perplexity_test.py

```diff
@@ -0,0 +1,345 @@
+# Copyright 2024 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import unittest
+import pytest
+
+from sharktank.evaluate import perplexity
+
+
+class PerplexityTest(unittest.TestCase):
+    def test_llama3_8B_f16_decomposed(self):
+
+        # Llama 3.1 8B decomposed
+
+        llama_8b_f16_gguf_path = "/data/extra/models/llama3.1_8B/llama8b_f16.gguf"
+        llama_8b_f16_tokenizer_path = (
+            "/data/extra/models/llama3.1_8B/tokenizer_config.json"
+        )
```

Comment:
Unit tests failed on this PR and after merge:
* https://github.com/nod-ai/SHARK-Platform/actions/runs/11359973189/job/31596941530
* https://github.com/nod-ai/SHARK-Platform/actions/runs/11360375745/job/31598023494

---

Any files required to run a test should be either
* included in the repository (if small enough)
* downloaded (and cached) on demand as part of the test
* downloaded (and cached) ahead of time via a script

As this is, this test will only run on a machine that has already run some unknown, undocumented setup steps.